### DISCUSSION TITLE
Background colour for systray icons

### DISCUSSION THREAD

**Discussion by skela** [qtile/qtile#3619] [General]
I have read the existing issues with the systray + previous discussions discussing theming of systray icons, still, I do wonder if there's a way to define a specific background colour for the individual icons that are displayed in the systray widget.

Why am I asking?

Well I've got a transparent background for my bar, and only colour in parts widgets to get a bar that's got different segments (see screenshot).

The systray icons seem to adopt the background colour of the bar (instead of the systray widget) causing the icons to have a transparent background colour, which looks a bit funky. 

Is there no way to have the systray icon draw in the background color for the widget below the icons? Alternatively add another property on the systray widget for icon background colour so it can be done? 

![2022-06-06_14-31](https://user-images.githubusercontent.com/58268/172163200-f3e7ca83-557a-4e16-ad10-24e143f8cfb3.png)

**Comment by elParaguayo** [comment]
I doubt this will be worked on. It's an issue in a lot of other window managers and none of them seem interested in working on it either, from what I can tell.

We discussed it when implementing transparency and felt it likely wasn't worth the effort (see https://github.com/qtile/qtile/pull/2513#issuecomment-860259864).

**Reply by skela** [reply]
Do you remember if any attempt was to add a layer below an icon in the draw method of the systray, where yer drawing the icons?
Any attempts at all in the draw method?

If so, what was the consequence, what worked / didnt work?

**Reply by elParaguayo** [reply]
> Do you remember if any attempt was to add a layer below an icon in the draw method of the systray,

For the "layer below" - the widget fills the whole space with the background colour before the icons are added. 

>  where yer drawing the icons?

SysTray doesn't draw icons. They are windows which are embedded into the bar. That's where most of the issues arise (compare the `StatusNotifier` widget which does draw icons).
